### PR TITLE
Implement a Gamepad drive command, etc.

### PR DESCRIPTION
**Pull request by andrewfhou**
Implement `GamepadDriveCommand` as well as other utility commands such as slow mode and reverse driving toggling

### CODE REVIEW

File: src/main/java/frc/team1816/robot/Robot.java [hunk 1]
```diff
@@ -31,7 +36,9 @@ public void disabledInit() { }
     public void autonomousInit() { }
 
     @Override
-    public void teleopInit() { }
+    public void teleopInit() {
+        Components.getInstance().drivetrain.setDefaultCommand(new GamepadDriveCommand());
```
Comment (pdeutsch) [inline src/main/java/frc/team1816/robot/Robot.java]: not a show stopper, but drivetrain should be private.  we should either be using getters/setters in Components or using Drivetrain.getInstance()

File: src/main/java/frc/team1816/robot/commands/ToggleReverseModeCommand.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+
+    /**
+     * The initialize method is called just before the first time
+     * this Command is run after being started.
+     */
+    @Override
+    protected void initialize() {
+        drivetrain.setReverseMode(!drivetrain.isReverseMode());
```
Comment (pdeutsch) [inline src/main/java/frc/team1816/robot/commands/ToggleReverseModeCommand.java]: or create toggleReverseMode in Drivetrain ...
code becomes ``` Components.getInstance().getDrivetrain().toggleReverseMode();```

File: src/main/java/frc/team1816/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -44,41 +50,76 @@ public Drivetrain() {
         this.rightSlaveOne = factory.getTalon(NAME, "rightSlaveOne", "rightMain");
         this.rightSlaveTwo = factory.getTalon(NAME, "rightSlaveTwo", "rightMain");
 
-        this.gyro = new PigeonIMU((TalonSRX) this.leftSlaveTwo);
+        this.gyro = new PigeonIMU((TalonSRX) this.leftSlaveTwo); // TODO: Determine whether using Talon breakout or CAN bus
```
Comment (pdeutsch) [inline src/main/java/frc/team1816/robot/subsystems/Drivetrain.java]: to be removed ...

File: src/main/java/frc/team1816/robot/Robot.java (hunk 1 above)
Comment (andrewfhou) [reply src/main/java/frc/team1816/robot/Robot.java]: If I recall, we ran into issues with that Singleton styling when trying to implement in for Zenith, but I'm not 100% sure. Will have to test it.

**Pull request merged**